libs/src/oracle-sdk-v2/services/oraclev1/ethers/factory.ts

```diff
@@ -0,0 +1,152 @@
+import Events from "events";
+import { ethers } from "ethers";
+import { assertAddress } from "@shared/utils";
+import type {
+  Request as SharedRequest,
+  OracleType,
+  ChainId,
+  RequestState as SharedRequestState,
+} from "@shared/types";
+import type { Address } from "wagmi";
+import type {
+  Handlers,
+  Service,
+  ServiceFactory,
+} from "@libs/oracle-sdk-v2/types";
+import type { TransactionReceipt } from "@libs/types";
+import type { Request } from "@libs/clients/optimisticOracle";
+import { RequestState, requestId } from "@libs/clients/optimisticOracle";
+import { OptimisticOracle } from "@libs/oracle-sdk-v1/services/optimisticOracle";
+
+export type Config = {
+  chainId: ChainId;
+  url: string;
+  address: string;
+};
+
+function convertToSharedState(state: RequestState): SharedRequestState {
+  if (state === RequestState.Invalid) return "Invalid";
+  if (state === RequestState.Requested) return "Requested";
+  if (state === RequestState.Proposed) return "Proposed";
+  if (state === RequestState.Expired) return "Expired";
+  if (state === RequestState.Disputed) return "Disputed";
+  if (state === RequestState.Resolved) return "Resolved";
+  return "Settled";
+}
+
+const ConvertToSharedRequest =
```

Comment:
tedious to write, but this ensures the data we compile through events matches the shared type expected in oracle data context